### PR TITLE
Fit plot improvements in MuonAnalysis: keep fit curves, set current data

### LINKED ISSUE #16017: Improvements to plot changes in MuonAnalysis
Improvements to the changes made recently in #15872 

Two points could be improved here:

1. Load a run, fit the data and then load another run on the Home tab. On going back to the Data Analysis tab, the dataset selected in the "Data" section is the old data, not the newly loaded run. This is a change from the previous behaviour and might be confusing.

2. Each new fit replaces the last, so there's only one fit curve on the graph at once. The original intention was to add each fit as a new curve, then you can use "clear fit curves" when the graph becomes cluttered.

### PR DESCRIPTION
**Pull request by tom-perkins**
Two improvements made to the recent changes in #15872:
- When the run is changed on the Home tab, the selected dataset to fit changes too. This returns to the previous behaviour before the recent changes.
- Fit curves remain on the graph when the run is changed (this is something asked for by scientists). They can be cleared using the "clear fit curves" option. This only applies to fits done in MuonAnalysis and _not_ elsewhere in Mantid.

**To test:**
_Example data at `\\olympic\babylon5\Scratch\TomPerkins\Data\MUSR`_
- Start the MuonAnalysis interface. On the "Settings" tab, set the _New plot policy_ to _Use previous window_.
- Load a datafile on the Home tab (type run number in or use Browse button)
- Go to "Data Analysis" tab and fit something to the dataset.
- Go back to Home tab and load the next run.  
  - The dataset on the graph should change, but the previous fit should remain. The plot difference should disappear.
- Go to "Data Analysis" tab again
  - The workspace in the "Data" section of the fit browser should be the new run you just loaded
- Run the fit on this new data
  - The new fit and old fit curves should both remain on the graph
- Click Display/Clear fit curves
  - Both fit curves should disappear from the graph

Fixes #16017.

[Release notes](https://github.com/mantidproject/mantid/blob/16017_MuonAnalysis_fit_plot_improvements/docs/source/release/v3.7.0/muon.rst#id3) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v3.7.0/muon.rst [hunk 1]
```diff
@@ -35,6 +35,7 @@ Muon Analysis
 
 - When the window is resized, all widgets within the window should now resize with it. This enables the interface to be used on smaller screens. `#15382 <https://github.com/mantidproject/mantid/pull/15832>`_
 - "Plot/Remove guess" now deals correctly with the case when a new run is loaded. `#15872 <https://github.com/mantidproject/mantid/pull/15872>`_
+- By default, previous fits now remain on the graph when a new run is loaded, to enable easy comparison between datasets. They can be removed with the "Clear fit curves" option. `#16018 <https://github.com/mantidproject/mantid/pull/16018>`_
```
Comment (KarlPalmen) [inline docs/source/release/v3.7.0/muon.rst]: This does not describe the functionality as you described in the test, which implies it is not by default, but only if you set *New plot policy* to *Use previous window*.

**Comment by KarlPalmen** [comment]
I've tested as describe and also when, *New plot policy* is set to *Use New Window* and then as expected the previous fit is not retained (nor is the previous window).

**Comment by KarlPalmen** [comment]
Muon Analysis Interface documentation apparently does not exist, so cannot be updated.

**Comment by KarlPalmen** [comment]
OK, if a small appropriate change is made to the release notes.

**Comment by tom-perkins** [comment]
Release notes amended - I've also updated the user documentation at http://www.mantidproject.org/MuonAnalysisSettings (thanks for reminding me!) This documentation should be in an rst file rather than the wiki but hasn't been moved yet - we have issue #13644 open to do this.

**Comment by KarlPalmen** [comment]
All is now good. :shipit: , subject to successful build.

**Pull request merged**